lib/DxcSupport/HLSLOptions.cpp

```diff
@@ -824,6 +824,17 @@ int ReadDxcOpts(const OptTable *optionTable, unsigned flagsToInclude,
   opts.VerifyDiagnostics = Args.hasFlag(OPT_verify, OPT_INVALID, false);
   if (Args.hasArg(OPT_ftime_trace_EQ))
     opts.TimeTrace = Args.getLastArgValue(OPT_ftime_trace_EQ);
+  if (Args.hasArg(OPT_ftime_trace_granularity_EQ)) {
+    if (Arg *A = Args.getLastArg(OPT_ftime_trace_granularity_EQ)) {
+      if (llvm::StringRef(A->getValue()).getAsInteger(10, opts.TimeTraceGranularity)) {
```

Comment:
Some of this could be simplified by using `getLastArgValue` upfront instead of checking for the arg, getting arg and then converting it to a stringref.